src/main/java/org/jetbrains/kotlinx/lincheck/annotations/Validate.kt

```diff
@@ -0,0 +1,41 @@
+/*-
+ * #%L
+ * Lincheck
+ * %%
+ * Copyright (C) 2019 - 2020 JetBrains s.r.o.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.jetbrains.kotlinx.lincheck.annotations
+
+import kotlin.annotation.AnnotationRetention.*
+import kotlin.annotation.AnnotationTarget.*
+
+/**
+ * It is possible in *lincheck* to add the testing data structure invariants validation,
+ * which is implemented via functions that can be executed multiple times during execution
+ * when there is no running operation in an intermediate state (e.g., in the stress mode
+ * they are invoked after each of the init and post part operations and after the whole parallel part).
+ * Thus, these functions should not modify the data structure.
+ *
+ * Validation functions should be marked with this annotation, do not have arguments,
```

Comment:
The same here:
Validation functions should be marked ... have no arguments and should not return anything